tools/model_tools/src/openvino/model_zoo/open_model_zoo.py

```diff
@@ -0,0 +1,217 @@
+                                        + str(precision) + '\'.\n    Allowed precision values for model are: '
+                                        + str(model.precisions))
+
+        prefix = Path(model_dir) / precision / model_name
+
+        model_path = str(prefix) + '.xml'
+        bin_path = str(prefix) + '.bin'
+        name = model.name
+        subdirectory = model.subdirectory
+
+        if not os.path.exists(model_path) or not os.path.exists(bin_path):
+            omz_converter.converter(['--name=' + model_name, '--precisions=' + precision,
+                        '--download_dir=' + str(download_dir)])
+        return cls(name, model_path, description, task_type, subdirectory, ie)
+
+    @classmethod
+    def from_pretrained(cls, model_path, *, task_type=None, ie=None):
+        '''
+        Loads model from existing .xml, .onnx files.
+        Parameters
+        ----------
+            model_path
+                Path to .xml or .onnx file.
+            task_type
+                Type of task that the model performs.
+            ie
+                Inference Engine instance
+        '''
+        model_path = Path(model_path).resolve()
+
+        if not model_path.exists():
+            raise ValueError('Path {} to model file does not exist.'.format(model_path))
+
+        if model_path.suffix not in ['.xml', '.onnx']:
+            raise ValueError('Unsupported model format {}. Only .xml or .onnx supported.'.format(model_path.suffix))
+
+        description = 'Pretrained model {}'.format(model_path.name)
+        name = model_path.stem
+        model = cls._load_model(cls, name)
+        task_type = model.task_type if model else task_type
+        subdirectory = model.subdirectory if model else model_path.parent
+        model_path = str(model_path)
+
+        return cls(name, model_path, description, task_type, subdirectory, ie)
+
+    def _load_model(self, model_name):
+        parser = argparse.ArgumentParser()
+        args = argparse.Namespace(all=False, list=None, name=model_name, print_all=False)
+        try:
+            model = _configuration.load_models_from_args(parser, args, _common.MODEL_ROOT,
+                        mode=_configuration.ModelLoadingMode.ignore_composite)[0]
+        except SystemExit:
+            model = None
+
+        return model
+
+    def _load(self, device):
+        self.net = self.ie.read_model(self.model_path)
+        self.exec_net = self.ie.compile_model(self.net, device)
+        self.device = device
+
+    def __call__(self, inputs, device='CPU'):
+        if self.ie is None:
+            raise TypeError('ie is not specified or is of the wrong type.'
+                            'Please check ie is of openvino.runtime.Core type.')
+
+        if self.exec_net is None or device != self.device:
+            self._load(device)
+
+        input_names = [input.get_any_name() for input in self.net.inputs]
+        infer_request = self.exec_net.create_infer_request()
```

Comment:
if it is not required for you to use the same infer request for each inference you can use
self.exec_net.infer_new_request, which combines request creation and inference inside

Reply:
Done